app/experimenter/experiments/serializers/design.py

```diff
@@ -198,32 +199,40 @@ def is_variant_valid(self, variants):
 
         return unique_names and non_empty
 
+    @transaction.atomic
```

Comment:
Do you have to use the decorator and the with statement? I think you only need one?